qiskit_nature/second_q/hamiltonians/lattices/kagome_lattice.py

```diff
@@ -0,0 +1,417 @@
+        return pos
+
+    def _style_pos(self) -> dict[int, list[float]]:
+        """Return a dictionary of positions for visualization of a two-dimensional lattice without
+            boundaries.
+
+        Returns:
+            dict[int, list[float]] : The keys are the labels of lattice points,
+                and the values are two-dimensional coordinates.
+        """
+        size = self._size
+        num_sites_per_cell = self._num_sites_per_cell
+        basis = self._basis
+        cell_positions = self._cell_positions
+        pos = {}
+
+        for cell_idx in range(np.prod(size)):
+            # maps an cell index to two-dimensional coordinate
+            # the positions are shifted so that the edges between boundaries can be seen
+            # for the periodic cases.
+            cell_coord = np.array(divmod(cell_idx, size[0])[::-1])
+
+            for i in range(num_sites_per_cell):
+                node_i = num_sites_per_cell * cell_idx + i
+                pos[node_i] = (np.dot(cell_coord, basis) + cell_positions[i]).tolist()
+        return pos
+
+    def __init__(
+        self,
+        rows: int,
+        cols: int,
+        edge_parameter: complex = 1.0,
+        onsite_parameter: complex = 0.0,
+        boundary_condition: BoundaryCondition = BoundaryCondition.OPEN,
+    ) -> None:
+        """
+        Args:
+            rows: Number of unit cells in the x direction.
+            cols: Number of unit cells in the y direction.
+            edge_parameter: Weight on all the edges, specified as a single value
+                Defaults to 1.0,
+            onsite_parameter: Weight on the self-loops, which are edges connecting a node to itself.
+                Defaults to 0.0.
+            boundary_condition: Boundary condition for the lattice.
+                The available boundary conditions are:
+                ``BoundaryCondition.OPEN``, ``BoundaryCondition.PERIODIC``.
+                Defaults to ``BoundaryCondition.OPEN``.
+        """
+        self._rows = rows
+        self._cols = cols
+        self._size = (rows, cols)
+        self._boundary_condition = boundary_condition
+        self._edge_parameter = edge_parameter
+        self._onsite_parameter = onsite_parameter
+
+        graph = PyGraph(multigraph=False)
+        graph.add_nodes_from(range(self._num_sites_per_cell * np.prod(self._size)))
+
+        # add edges excluding the boundary edges
+        bulk_edges = self._bulk_edges()
+        graph.add_edges_from(bulk_edges)
+
+        # add self-loops
+        self_loop_list = self._self_loops()
+        graph.add_edges_from(self_loop_list)
+
+        # add edges that cross the boundaries
+        boundary_edge_list = self._boundary_edges()
+        graph.add_edges_from(boundary_edge_list)
+
+        # a list of edges that depend on the boundary condition
+        self.boundary_edges = [(edge[0], edge[1]) for edge in boundary_edge_list]
```

Comment:
This ` self.boundary_edges` looks like a public instance variable - is that intended - do we expect this to be part of a public API? Elsewhere property getters are used to expose private instance vars as readonly. I can see this used in the draw without boundary method - if this is the intent perhaps it should just be a private instance var.

Reply:
Yes, this one I am also not sure about. I think we should make it internal.
If (in the future) we find the need for it to be public we can always add property getter/setter methods around it.

In the draw method we can easily just use the private attribute since its within the same class